### DISCUSSION TITLE
Tally up of the number of cards that were passed to you in their colors, ideally only count cards that fit certain condition (No Gold cards, GIH WR > 55% for example) to identify potential open lanes

### DISCUSSION THREAD

**Discussion by vudinh29** [unrealities/MTGA_Draft_17Lands#56] [Ideas]
The idea is to keep a count of good/playable cards that were passed to you, pick after pick, to get a sense of what color is open as soon as possible. P1P1, P1P2 and P1P3 we just don't count anything, because there is not yet any signal.
If effort is too much on this, I would love some pointers to the code base as to what sections I can start looking at to implement this is a "side" feature that doesn't change anything else about the app. I'm not a SWE, just a data guy that uses Python so it would be hard but I can try.

**Comment by unrealities** [comment]
I was thinking about a similar feature. Finding a suitable algorithm with play boosters makes this even more complicated, as fewer clear signals exist. Seeing an uncommon signpost at P1P4 may indicate an open pair or may mean there were three playable rares.

Creating a simple +/- scoring system for ALSA vs. the current pick may be an acceptable initial pass.

[src/log_scanner.py](../blob/main/src/log_scanner.py) has a lot of core logic around the picks made, so the code would go there, and testing would be done in [tests/test_log_scanner.py](../blob/main/tests/test_log_scanner.py).

I don't know where best to display the data visually. That would occur in [src/overlay.py](../blob/main/src/overlay.py).

**Comment by pldefg005** [comment]
Did this go anywhere? I had the same idea the other day and figured I'd come here to see if there was a way to do this. Alternately, I was thinking if I could quickly "export" the data myself for a given pack while drafting, I could quickly dump that somewhere for tracking purposes.

**Comment by unrealities** [comment]
I made a simple implementation of this in version 3.37. It could probably be improved in multiple ways.

https://github.com/unrealities/MTGA_Draft_17Lands?tab=readme-ov-file#signal-detection-beta
